src/main/java/seedu/address/logic/commands/GemCommand.java

```diff
@@ -0,0 +1,119 @@
+        sem1Storage.setGehModules(GEH_SEM1_PATH);
+        sem1Storage.setGeqModules(GEQ_PATH);
+        sem1Storage.setGerModules(GER_PATH);
+        sem1Storage.setGesModules(GES_SEM1_PATH);
+        sem1Storage.setGetModules(GET_SEM1_PATH);
+    }
+
+    /**
+     * Loads the sem2Storage attribute with Semester 2 GE Modules by using
+     * the various set() methods from the GemCommandStorage class.
+     * @throws IOException When the path in invalid.
+     * @throws IllegalValueException When the data from the JSON file does not match the
+     * specific field headers of the JsonAdaptedModule class (Eg.'moduleCode', 'modularCredits').
+     */
+    public void setSem2Storage() throws IOException, IllegalValueException {
+        sem2Storage = new GemCommandStorage();
+        sem2Storage.setGehModules(GEH_SEM2_PATH);
+        sem2Storage.setGeqModules(GEQ_PATH);
+        sem2Storage.setGerModules(GER_PATH);
+        sem2Storage.setGesModules(GES_SEM2_PATH);
+        sem2Storage.setGetModules(GET_SEM2_PATH);
+    }
+
+    /**
+     * Takes a List of Modules and extracts out their Module Code and Modular Credits.
+     * @param modules List of Modules.
+     * @return String of Module Codes and Modular Credits.
+     */
+    public StringBuilder moduleExtractor(ObservableList<Module> modules) {
+        StringBuilder modulesToAdd = new StringBuilder();
+        for (Module module : modules) {
+            String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+            modulesToAdd.append("\n").append(moduleToAdd);
+        }
+        return modulesToAdd;
+    }
+
+    /**
+     * Sets up the all the GE modules in the sem1Storage and sem2Storage and displays them
+     * with the CommandResult object.
+     * @param model {@code Model} which the command should operate on.
+     * @return a CommandResult displaying all the available GE modules.
+     */
+    @Override
+    public CommandResult execute(Model model) {
+        try {
+            requireNonNull(model);
+            setSem1Storage();
+            setSem2Storage();
+            String modulesToAdd = "\n \n" + "SEMESTER 1:" + "\n";
+            modulesToAdd += moduleExtractor(sem1Storage.getGehModules());
+            modulesToAdd += moduleExtractor(sem1Storage.getGeqModules());
+            modulesToAdd += moduleExtractor(sem1Storage.getGerModules());
+            modulesToAdd += moduleExtractor(sem1Storage.getGesModules());
+            modulesToAdd += moduleExtractor(sem1Storage.getGetModules());
+            modulesToAdd += "\n \n" + "SEMESTER 2:" + "\n";
+            modulesToAdd += moduleExtractor(sem2Storage.getGehModules());
+            modulesToAdd += moduleExtractor(sem2Storage.getGeqModules());
+            modulesToAdd += moduleExtractor(sem2Storage.getGerModules());
+            modulesToAdd += moduleExtractor(sem2Storage.getGesModules());
+            modulesToAdd += moduleExtractor(sem2Storage.getGetModules());
```

Comment:
Suggestion: should we add a `getAllModules` method to `GemCommandStorage` instead and call that?

Reply:
Yup, to make it cleaner, I made a new method called setCompiledModules() in the GemCommandStorage class!